### DISCUSSION TITLE
Pressure Advance / K Factor settings, which one is used?

### DISCUSSION THREAD

**Discussion by barsk** [OrcaSlicer/OrcaSlicer#2675] [Q&A] [answered]
Hi, for a Bambu printer (P1S) with Bambu Studio the settings for K Factor is in the device tab under spool/AMS where you can push the pen (edit) icon and enter a K factor, e.g 0.020. There is no way to store the settings for K factor (or Pressure advance) anywhere in Bambu Studio for a custom filament as far as I know. For the Bambu filaments I believe it is stored in firmware. So, as I understand it you have to edit the K Factor _each_ time you change a spool with Bambu Studio in the device tab.

Now to my question, in Orca we can set Pressure advance (same thing as K factor) in the filament (advanced) settings. BUT, there is _also_ the setting for K factor still in the device tab in Orca. So, which one is used and why is there two settings?

**Comment by LJKraus** [comment]
It would be nice to have a definitive answer to this question.  For now, I rely solely on the PA setting in the filament preset, not entering anything in the Device tab.  Results have continued to be good.   But the question remains,  if the "Studio calibration" number is different, could it override the "Orca setting"?   Or is the Device tab K factor only used with files sliced by Studio/Makerworld?

**Comment by SoftFever** [comment, marked as answer]
Pressure advance set in the filament setting will overwrite the K value in slot

**Reply by barsk** [reply]
Ok, got it. This could maybe find it's way to the wiki.

However, will a value of 0.000 in the device slot mean that pressure advance is _disabled_ as well. If we have PA disabled in settings and 0.000 in device slot (K value), would that mean that PA is _enabled_ with a value of 0.000? Or is it the same thing...?

I'm asking for the purpose of setting up for a calibration. What is the proper preconditions for a pressure advance calibration? Disable in settings and enter 0.000 in slot, or do those settings/numbers get overriden in the calibration gcode whatever they are?

**Reply by SoftFever** [reply]
`will a value of 0.000 in the device slot mean that pressure advance is disabled as well.`
I don't think so. However, it's no harm to test it by yourself if you're in doubt. For instance, try setting a large PA value, like 0.1, to see what happens.

**Reply by barsk** [reply]
Hmm, I think you misunderstood. What if I set PA as disabled, and K value = 0.000. Will that be _used_ as PA enabled = 0.000? or will it be _used_ as PA = disabled?

There is no setting for disabling K value in the slot section, and setting 0.000 seems like the way it is resetted in bambu studio. It is a bit confusing.
